modules/heart/heart.js

```diff
@@ -0,0 +1,64 @@
+import Component from "@pencil.js/component";
+
+/**
+ * Heart class
+ * @class
+ * @extends Component
+ */
+export default class Heart extends Component {
+    /**
+     * Heart constructor
+     * @param {PositionDefinition} positionDefinition - Position of the center of the heart
+     * @param {Number} radius - radius The distance from center of heart
+     * @param {ComponentOptions} [options] - Drawing options
+     */
+    constructor (positionDefinition, radius, options) {
+        super(positionDefinition, options);
+
+        if (radius > 0) {
+            this.radius = radius;
+        }
+        else {
+            this.radius = 1;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    toJSON () {
+        const json = super.toJSON();
+        delete json.radius;
+        return json;
+    }
```

Comment:
Here, you should be adding `radius` to the result from the parent. Take a look at what Rectangle is doing for example.